subsys/bluetooth/host/Kconfig.gatt

```diff
@@ -202,6 +202,15 @@ config BT_DEVICE_NAME_GATT_WRITABLE
 	  Enabling this option allows remote GATT clients to write to device
 	  name GAP characteristic.
 
+if BT_DEVICE_NAME_GATT_WRITABLE
+config BT_DEVICE_NAME_GATT_WRITABLE_WITHOUT_BONDING
```

Comment:
It is misleading to say without bonding, because bonding is a different concept.

I would suggest to align the Kconfig names with the gatt permission definitions, also I think we should make this complete by having configuration for authentication permission as well.

When a similar kconfig was added for the SMP_SVR service it was named `CONFIG_MCUMGR_SMP_BT_AUTHEN`

I would then suggest the following:

```
DEVICE_NAME_GATT_WRITABLE_ENCRYPT
    default y

DEVICE_NAME_GATT_WRITABLE_AUTHEN
    # default n to keep behavior the same
```

Reply:
@HuangJunMIOT could you please reply to the review comment?